tests/psa-client-server/psasim/src/psa_sim_generate.pl

```diff
@@ -1093,15 +1103,48 @@ sub get_functions
     my %funcs = ();
     for (my $i = 0; $i <= $#src; $i++) {
         my $line = $src[$i];
-        if ($line =~ /^(psa_status_t|uint32_t|void) (psa_\w*)\(/) { # begin function definition
+        if ($line =~ /^(static(?:\s+inline)?\s+)?
+                       ((?:(?:enum|struct|union)\s+)?\w+\s*\**\s*)\s+
+                       ((?:mbedtls|psa)_\w*)\(/x) {
+            # begin function declaration
             #print "have one $line\n";
             while ($line !~ /;/) {
                 $line .= $src[$i + 1];
                 $i++;
             }
+            if ($line =~ /^static/) {
+                # IGNORE static inline functions: they're local.
+                next;
+            }
             $line =~ s/\s+/ /g;
             if ($line =~ /(\w+)\s+\b(\w+)\s*\(\s*(.*\S)\s*\)\s*[;{]/s) {
                 my ($ret_type, $func, $args) = ($1, $2, $3);
+
+                if ($func =~ /_init$/ and $func ne 'psa_crypto_init' and
+                    $args eq 'void') {
+                    # IGNORE constructors
+                    next;
+                }
+                if ($func =~ /_pake_/ or
```

Comment:
There's an array of `skip_functions` at the top to avoid having to put all these things in the body, and anything in that list gets removed from what is parsed.

I think I'd rather have the parsing code separate from the "we ignore function X for reason Y"s

Reply:
Ah, thanks. I don't understand the data flow, but whatever works.